src/game/Game.test.ts

```diff
@@ -38,11 +39,66 @@ describe("GameState", () => {
         });
 
         test("picking outside of range fails", () => {
-            const game = new Game(threeByThreeMatrix);
+            const game = new Game(threeByThreeMatrix, [], unrestrictedBuffer);
             expect(() => game.pick(-1, 0)).toThrow();
             expect(() => game.pick(0, -1)).toThrow();
             expect(() => game.pick(3, 0)).toThrow();
             expect(() => game.pick(0, 3)).toThrow();
         });
+
+        test("picking fills buffer, fulfills sequence", () => {
+            const simpleSequence = ["00", "10", "20"]
+            const game = new Game(threeByThreeMatrix, [simpleSequence], unrestrictedBuffer);
+            expect(game.getSequences()).toEqual([{sequence: simpleSequence, numberOfFulfilled: 0}])
+            game.pick(0, 0);
+            expect(game.buffer).toEqual(["00"]);
+            expect(game.getSequences()).toEqual([{sequence: simpleSequence, numberOfFulfilled: 1}])
+        });
+
+        test("picking fulfills second sequence occurence", () => {
+            const sequence = ["AA", "BB", "CC"]
+            const game = new Game([
+                    "AA", "AA", "BB",
+                    "BB", "CC", "AA",
+                    "CC", "CC", "CC",
+                ], [sequence], unrestrictedBuffer);
+            expect(game.getSequences()).toEqual([{sequence: sequence, numberOfFulfilled: 0}])
+            game.pick(0, 0);
+            expect(game.getSequences()).toEqual([{sequence: sequence, numberOfFulfilled: 1}])
+            game.pick(2, 0);
+            expect(game.getSequences()).toEqual([{sequence: sequence, numberOfFulfilled: 2}])
+            game.pick(2, 1);
+            expect(game.getSequences()).toEqual([{sequence: sequence, numberOfFulfilled: 1}])
+            game.pick(0, 1);
+            expect(game.getSequences()).toEqual([{sequence: sequence, numberOfFulfilled: 2}])
+            game.pick(0, 2);
+            expect(game.getSequences()).toEqual([{sequence: sequence, numberOfFulfilled: 3}])
+        });
+    });
+
+    describe("game end conditions", () => {
+        test("game won", () => {
+            const game = new Game([
+                    "AA", "AA", "BB",
+                    "BB", "CC", "AA",
+                    "CC", "CC", "CC",
+                ], [["AA", "BB", "CC"]], 3)
+            game.pick(0, 0)
+            game.pick(2, 0)
+            game.pick(2, 2)
+            expect(game.state).toEqual(EndState.Won)
+        })
+
+        test("game loose", () => {
+            const game = new Game([
+                    "AA", "AA", "BB",
+                    "BB", "CC", "AA",
+                    "CC", "CC", "CC",
+                ], [["AA", "BB", "CC"]], 3)
+            game.pick(0, 0)
+            game.pick(1, 0)
+            game.pick(1, 2)
+            expect(game.state).toEqual(EndState.Lost)
+        })
```

Comment:
Test to make sure pick does not work after the buffer is full.

Reply:
done